proto/substrait/algebra.proto

```diff
@@ -52,6 +53,22 @@ message RelCommon {
       substrait.extensions.AdvancedExtension advanced_extension = 10;
     }
   }
+
+  // Optional metadata that the producer may attach to the relation.
+  // Consumer should hold no expectations about the existance metadata.
+  message Metadata {
+    // Name (alias) for this relation.
+    // If provided, must be unique across the relations in the plan/root.
```

Comment:
Technically it only needs to be unique within the subplan that consumes it.  But this is a safer qualification.

Reply:
I hate to bike shed on names but if it must be unique across all relations then would `id` be a better name to use here than `alias`?

Reply:
I'd have a slight preference for "alias" rather than "id", as "alias" is the name used in SQL (eg. https://www.w3schools.com/sql/sql_alias.asp) and also "id" sounds like something people might expect to be able to refer to later on in the (substrait) plan, which we probably don't want. But I can be convinced otherwise :D

Reply:
Let's stick with alias then.